x-pack/plugins/security_solution/server/endpoint/services/actions/create/index.ts

```diff
@@ -76,7 +77,12 @@ export const actionCreateService = (
 
   return {
     createAction,
-    createActionFromAlert: (payload, agents) =>
-      createAction(payload, agents, { minimumLicenseRequired: 'enterprise' }),
+    createActionFromAlert: async (payload) => {
+      const endpointData = await endpointContext.service
+        .getEndpointMetadataService()
+        .getMetadataForEndpoints(esClient, [...new Set(payload.endpoint_ids)]);
```

Comment:
1. Computing unique endpoint_ids is redundant here as the `payload.endpoint_ids` aways has a single id.
2. You can pass in the esClient to the `getScheduleNotificationResponseActionsService` at plugin start and then use that to pass in the unique `agentIds` wher `createActionFromAlert` is called. 
3. Maybe even better if pass in the alerts and unique agentIds to the `getScheduleNotificationResponseActionsService.endpointResponseAction` method itself.

Reply:
Hey, thanks for checking this 👍 
We do send multiple endpoint_ids in payload from alert, so we could show an error of missing integration. So basically we make a call on multiple agents, but in create action we verify for only the ones with integration and then return alert. 
How does this sound?

Reply:
Sorry for answering in 2 pieces:
I verified, the difference is that we want to have endpoint_ids in payload to use it here:https://github.com/elastic/kibana/blob/160fa420ef7d32a89c965806d047f3e091475687/x-pack/plugins/security_solution/server/endpoint/services/actions/create/write_action_to_indices.ts#L52 
but firstly we get agentIds (the ones with integration installed) out of that endpoint_ids, and if that doesn't have integration - we return error message in action response. 
That is why I think having both endpoint_ids and agents is fine. I think the same is happening with any other usages of createAciton, eg. cases and file_upload, isnt it?

Reply:
Also: re the unique list of ids. It's just to be aligned with other places where we do it, because we are actually capable of sending multiple endpoint_ids and calling it at once, right?